### PR TITLE
Run crypto integration tests on every PR

### PR DESCRIPTION
**Pull request by Anderas**
Enable selected (currently non-failing) integration tests related to the crypto module to be run on every PR in parallel to unit tests. Whilst integration tests could run nightly or even after PR merge, encryption is important enough to test as soon as possible, and given that it can run in parallel to unit tests and takes similar amount of time, it should not increase the overal PR checks by much.

### CODE REVIEW

**Review by gileluard — APPROVED** [review]
LGTM but @manuroe may have some feedback.

**Review by bmarty — COMMENTED** [review]
I do not know why I have a review request on this PR.
Please ignore my comment.

**Review by bmarty — COMMENTED** [review]
(please ignore this comment)

File: MatrixSDK/MXRestClient.m [hunk 1]
```diff
@@ -266,19 +266,23 @@ -(id)initWithCredentials:(MXCredentials*)inCredentials
                         dispatch_async(self.completionQueue, ^{
                             BOOL isSoftLogout = [MXRestClient isSoftLogout:mxError];
                             MXLogDebug(@"[MXRestClient] tokenProviderHandler: %@: non-refresh(access token) token auth failed", logId);
-                            self.unauthenticatedHandler(mxError, isSoftLogout, NO, ^{
-                                failure(error);
-                            });
+                            if (unauthenticatedHandler) {
```
Comment (manuroe) [inline MatrixSDK/MXRestClient.m:269]: Thanks. I saw this problem last Thursday

File: MatrixSDKTests/MXCryptoTests.m [hunk 2]
```diff
@@ -3142,11 +3141,9 @@ - (void)testEncryptionAlgorithmChange
                 XCTAssertEqual(liveTimeline.state.encryptionAlgorithm.length, 0);   // with a nil algorithm
                 XCTAssertTrue(roomFromAlicePOV.summary.isEncrypted);
                 
-                // -> It must be impossible to send a messages (because the algorithm is not supported)
+                // -> It is still possible to send a message because crypto will use backup algorithm (which can never be removed)
```
Comment (manuroe) [inline MatrixSDKTests/MXCryptoTests.m:3144]: Can I ask if this is the same new behavior on android and web?
I am happy with the change. I just want to check the consistency.

**Review by manuroe — COMMENTED** [review]
Thanks. It will be super useful.
I just have a question.

File: MatrixSDKTests/MXCryptoTests.m (hunk 2 above)
Comment (manuroe) [reply MatrixSDKTests/MXCryptoTests.m:3144]: As far as I understand the convo in the crypto room, this is fine if the SDK continues to encrypt messages using the previously defined algorithm. This is the app that should alarm the end user about the wrong encryption configuration (https://github.com/vector-im/element-ios/issues/5322).

I am now happy about this test change.

**Review by manuroe — APPROVED** [review]

**Pull request merged**